src/Entity/PrintArticleType.php

```diff
@@ -228,21 +228,25 @@ public function validate() {
     $idmsViolations = $idms->validate();
 
     // Validate $this.
-    $printArticleTypeViolations = Validation::createValidator()->validate($this, [
-      new IdmsUniqueTags(),
-    ]);
+    $printArticleTypeViolations = Validation::createValidatorBuilder()
+      ->addMethodMapping('loadValidatorMetadata')
+      ->getValidator()
+      ->validate($this);
 
-    $tagViolations = Validation::createValidator()->validate($this->getTags(), [
-      new NotBlank(),
-    ]);
+    $printArticleTypeViolations->addAll($idmsViolations);
 
-    $violationList = new ConstraintViolationList();
-    $violationList->addAll($idmsViolations);
-    $violationList->addAll($printArticleTypeViolations);
-    $violationList->addAll($tagViolations);
-
-    return $violationList;
+    return $printArticleTypeViolations;
+  }
 
+  /**
+   * Provides metadata for validator.
+   *
+   * @param \Symfony\Component\Validator\Mapping\ClassMetadata $metadata
+   *   Symfony valdiator metadata object.
+   */
+  public static function loadValidatorMetadata(ClassMetadata $metadata) {
+    $metadata->addGetterMethodConstraint('mapping', 'getTags', new NotBlank(['message' => "IDMS doesn't contain defined tags from the tag-mapping."]));
```

Comment:
With using `idms` instead of `mapping` here, you will operate on the real property and the form will automatically mark the file upload as invalid: see http://share.undpaul.de/bilder/jh/Add_snippet_template__Site-Install_2017-05-02_12-51-24--89ygr.png